packages/invoices/src/InboundShipment/DetailView/api.ts

```diff
@@ -204,3 +218,121 @@ export const getInboundShipmentDetailViewApi = (
     throw new Error(':shrug');
   },
 });
+
+export const useInboundShipment = (): UseQueryResult<Invoice, unknown> => {
+  const { id } = useParams();
+  const { api } = useOmSupplyApi();
+  const queries = getInboundShipmentDetailViewApi(api);
+  return useQuery(['invoice', id], () => {
+    return queries.onRead(id);
+  });
+};
+
+export const useInboundShipmentSelector = <T = Invoice>(
+  select?: (data: Invoice) => T
+): UseQueryResult<T, unknown> => {
+  const { id } = useParams();
+  const { api } = useOmSupplyApi();
+  const queries = getInboundShipmentDetailViewApi(api);
+  return useQuery(
+    ['invoice', id],
+    () => {
+      return queries.onRead(id);
+    },
+    { select }
+  );
+};
+
+const useOptimisticInboundUpdate = () => {
+  const { api } = useOmSupplyApi();
+  const queries = getInboundShipmentDetailViewApi(api);
+  const queryClient = useQueryClient();
+  const { id } = useParams();
+  return useMutation(queries.onUpdate, {
+    onMutate: async (patch: Partial<InboundShipment>) => {
+      await queryClient.cancelQueries(['invoice', id]);
+
+      const previousInbound: Invoice = queryClient.getQueryData([
+        'invoice',
+        id,
+      ]);
+
+      queryClient.setQueryData(['invoice', id], {
+        ...previousInbound,
+        ...patch,
+      });
+
+      return { previousInbound, patch };
+    },
+    onSettled: () => queryClient.invalidateQueries(['invoice', id]),
+    onError: (_, __, context) => {
+      queryClient.setQueryData(['invoice', id], context.previousInbound);
+    },
+  });
+};
+
+export const useInboundLines = () => {
+  const { sortBy, onChangeSortBy } = useSortBy<OutboundShipmentSummaryItem>({
+    key: 'itemName',
+  });
+  const onSort = (column: Column<OutboundShipmentSummaryItem>) => {
+    onChangeSortBy({
+      key: column.key,
+      isDesc: sortBy.key === column.key ? !sortBy.isDesc : false,
+    });
+  };
+  const selectItems = useCallback(
+    (invoice: Invoice) => {
+      return inboundLinesToSummaryItems(invoice.lines).sort(
+        getDataSorter(
+          sortBy.key as keyof OutboundShipmentSummaryItem,
+          !!sortBy.isDesc
+        )
+      );
+    },
+    [sortBy]
+  );
+
+  const { data } = useInboundShipmentSelector(selectItems);
+
+  return { data, sortBy, onSort };
+};
+
+export const useDraftInbound = () => {
```

Comment:
I'm fine with continuing to use `draft` even though the status is `NEW` now. the term draft still makes sense, don't think there's a need to go changing it all..